### PR TITLE
Feat Improvements to Activities

### LINKED ISSUE #129: Display Activities in chronological order
For now activities aren't displayed in chronological order on the ByDay and Activities Screen. They should be ordered by the start time (and date), and this case this is equal, by the end time

### PR DESCRIPTION
**Pull request by ioanajianu**
## Summary

This change improves some aspects that relate to activities:
1. Activities screen uses the trip activities instead of hardcoded values
2. Draft activities don't appear on the ByDay screen (before they were appearing as 1970/01/01)
3. Activities are now displayed in chronological order, both in the ByDay screen and Activities screen
4. The Activities screen now has an 'Add Activity' button 
5. Added tests

## Checklist

Ensure all / most of these are checked before the pull request is submitted.
- [x] This PR resolves #129 
- [x] Tests have been added for new functionality.
- [ ] Screenshots or UI changes have been attached (if applicable).
- [ ] Documentation has been updated (if applicable).

##  Testing

Explain how this feature was tested and what kind of testing (unit, integration, UI) has been done. Mention platforms or configurations used for testing.
- **Manual Testing:**
    - [x] Tested on emulator / physical device.
- **Unit Tests**
    - [x] Added units tests for this

If you haven't added tests, please create a new issue and assign it.

##  Related Issues/Tickets

#130

### CODE REVIEW

File: app/src/main/java/com/android/voyageur/ui/trip/schedule/ByDay.kt [hunk 1]
```diff
@@ -68,77 +66,19 @@ fun ByDayScreen(
...
-                Activity(
-                    title = "Visit city",
-                    description = "",
-                    activityType = ActivityType.OTHER,
-                    startTime = oneDayAfterNow,
-                    endTime = Timestamp.now(),
-                    estimatedPrice = 00.00,
-                    location = Location()))
-        val groupedActivities = groupActivitiesByDate(tripActivities)
+        val groupedActivities =
```
Comment (1andreea26) [inline app/src/main/java/com/android/voyageur/ui/trip/schedule/ByDay.kt:71]: Thanks for replacing and sorting this in ByDay screen!

File: app/src/main/java/com/android/voyageur/ui/trip/schedule/ByDay.kt [hunk 2]
```diff
@@ -166,7 +106,14 @@ fun ByDayScreen(
 
 // TODO: Test this works
 fun groupActivitiesByDate(activities: List<Activity>): Map<LocalDate, List<Activity>> {
-  return activities.groupBy { activity ->
+  val sortedActivities =
```
Comment (1andreea26) [inline app/src/main/java/com/android/voyageur/ui/trip/schedule/ByDay.kt:109]: And thanks again for updating this function :)

**Review by 1andreea26 — APPROVED** [review]
I will approve this as I don't have anything to add. Code is nicely written,  comments help with understanding the changes, and line coverage is reached. I suggested just one test if you want to add it. Good job!

File: app/src/androidTest/java/com/android/voyageur/ui/trip/activities/ActivitiesScreenTest.kt [hunk 3]
```diff
@@ -78,11 +91,12 @@ class ActivitiesScreenTest {
```
[the remark this answers is not in the document]
Comment (ioanajianu) [reply app/src/androidTest/java/com/android/voyageur/ui/trip/activities/ActivitiesScreenTest.kt:93]: Thanks for the suggestion! I added the test

**Pull request merged**